File: src/p0003/cpp/solution.cpp
class Solution {
public:
    static int lengthOfLongestSubstring(const string &s) {
        int result = 0;
        vector<int> count(256, 0);

        for (vector<int>::size_type i = 0, j = 0; j < s.length(); j++) {
            for (count[s[j]]++; count[s[j]] > 1; count[s[i]]--, i++);
            result = max(result, static_cast<int>(j - i + 1));
        }

        return result;
    }
};

File: src/p0003/golang/solution.go
func lengthOfLongestSubstring(s string) int {
	result := 0
	count := make([]int, 256)

	for i, j := 0, 0; j < len(s); j++ {
		for count[s[j]]++; count[s[j]] > 1; i++ {
			count[s[i]]--
		}

		if result < j+1-i {
			result = j + 1 - i
		}
	}

	return result
}
